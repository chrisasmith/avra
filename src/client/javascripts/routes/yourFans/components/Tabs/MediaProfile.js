import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { injectIntl, defineMessages } from 'react-intl'
import styles from '../styles/YourFans.scss'
import * as yourFansActions from '../../../../reducers/pages/yourFansActions'
import { createLoadingSelector } from '../../../../reducers/selectors'
import DashboardGraphModule, { MODULE_TYPE } from '../../../../components/GridCards/DashboardGraphModule'
import FilterBreadCrumb from '../../../../components/Filter/FilterBreadCrumb'

const propTypes = {
  data: PropTypes.object,
  filter: PropTypes.object,
  isFetching: PropTypes.bool.isRequired,
  intl: PropTypes.object,
}

const messages = defineMessages({
  premierLeague: {
    id: 'your_fans.tab.media.premier_league',
    defaultMessage: 'Interacting with Premier League',
  },
  useToFollow: {
    id: 'your_fans.tab.media.use_to_follow',
    defaultMessage: 'Which of the following do you use to interact with or follow the Premier League or your favorite Premier League team?',
  },
  activities: {
    id: 'your_fans.tab.media.activities',
    defaultMessage: 'Activities',
  },
  sportWatching: {
    id: 'your_fans.tab.media.watching',
    defaultMessage: 'Sport Watching',
  },
  whileWatching: {
    id: 'your_fans.tab.media.while_watching',
    defaultMessage: 'While Watching Sports',
  },
  howOften: {
    id: 'your_fans.tab.media.how_often',
    defaultMessage: 'How often do you do the following?',
  },
})

const MediaProfileTab = injectIntl(({ intl, filter, data, isFetching }) => (
  <div className={styles.grid}>
    <div className={styles.breadCrumb}>
      <FilterBreadCrumb filter={filter} />
    </div>
    <div className="container">
      <div className="row row-grid">
        <div className="col-6">
          <DashboardGraphModule
            title={intl.formatMessage(messages.premierLeague)}
            description={intl.formatMessage(messages.useToFollow)}
            sideTitle={intl.formatMessage(messages.activities)}
            data={data ? data.media_interacting : null}
            isFetching={isFetching}
            type={MODULE_TYPE.PERCENT}
            propertyTitle={filter.property.name}
            benchmarkTitle={filter.benchmark.name}
          />

          <DashboardGraphModule
            title={intl.formatMessage(messages.sportWatching)}
            data={data ? data.sport_media : null}
            isFetching={isFetching}
            type={MODULE_TYPE.PERCENT}
            propertyTitle={filter.property.name}
            benchmarkTitle={filter.benchmark.name}
          />
        </div>
        <div className="col-6">
          <DashboardGraphModule
            title={intl.formatMessage(messages.whileWatching)}
            description={intl.formatMessage(messages.howOften)}
            data={data ? data.while_watching : null}
            isFetching={isFetching}
            type={MODULE_TYPE.PERCENT}
            propertyTitle={filter.property.name}
            benchmarkTitle={filter.benchmark.name}
          />
        </div>
      </div>
    </div>
  </div>
))

function mapStateToProps(state) {
  const loadingSelector = createLoadingSelector([yourFansActions.GET_HOME_PAGE.SELF])
  const filter = state.filters.filter
  const data = state.pages.yourFans.data
  return { ...state, isFetching: loadingSelector(state), filter, data }
}

MediaProfileTab.propTypes = propTypes
export default connect(mapStateToProps)(MediaProfileTab)
