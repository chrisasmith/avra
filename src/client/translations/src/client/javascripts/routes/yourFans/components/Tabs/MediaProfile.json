[
  {
    "id": "your_fans.tab.media.premier_league",
    "defaultMessage": "Interacting with Premier League"
  },
  {
    "id": "your_fans.tab.media.use_to_follow",
    "defaultMessage": "Which of the following do you use to interact with or follow the Premier League or your favorite Premier League team?"
  },
  {
    "id": "your_fans.tab.media.activities",
    "defaultMessage": "Activities"
  },
  {
    "id": "your_fans.tab.media.watching",
    "defaultMessage": "Sport Watching"
  },
  {
    "id": "your_fans.tab.media.while_watching",
    "defaultMessage": "While Watching Sports"
  },
  {
    "id": "your_fans.tab.media.how_often",
    "defaultMessage": "How often do you do the following?"
  }
]