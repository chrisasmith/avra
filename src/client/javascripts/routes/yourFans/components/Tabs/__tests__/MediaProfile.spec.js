import React from 'react'
import { shallow } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { shallowToJson } from 'enzyme-to-json'
import MediaProfile from '../MediaProfile'

const mockStore = configureMockStore([thunk])

let store

const getTab = s => shallow(<MediaProfile store={mockStore(s)} />).shallow()

describe('Media Profile Tab', () => {
  beforeEach(() => {
    store = {
      location: { },
      loading: {

      },
      error: {

      },
      filters: {
        filter: {
          property: {
            name: 'property',
          },
          benchmark: {
            name: 'benchmark',
          },
        },
      },
      pages: {
        yourFans: {
          data: null,
        },
      },
    }
  })

  it('should show', () => {
    const wrapper = getTab(store)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should show loading', () => {
    store.loading.GET_HOME_PAGE = true

    const wrapper = getTab(store)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
