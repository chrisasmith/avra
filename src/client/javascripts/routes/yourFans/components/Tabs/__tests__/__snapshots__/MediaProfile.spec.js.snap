// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media Profile Tab should show 1`] = `
<div
  className="grid"
>
  <div
    className="breadCrumb"
  >
    <FilterBreadCrumb
      filter={
        Object {
          "benchmark": Object {
            "name": "benchmark",
          },
          "property": Object {
            "name": "property",
          },
        }
      }
    />
  </div>
  <div
    className="container"
  >
    <div
      className="row row-grid"
    >
      <div
        className="col-6"
      >
        <DashboardGraphModule
          benchmarkTitle="benchmark"
          data={null}
          description="Which of the following do you use to interact with or follow the Premier League or your favorite Premier League team?"
          isFetching={false}
          propertyTitle="property"
          sideTitle="Activites"
          title="Interacting with Premier League"
          type="PERCENT"
        />
        <DashboardGraphModule
          benchmarkTitle="benchmark"
          data={null}
          isFetching={false}
          propertyTitle="property"
          title="Sport Watching"
          type="PERCENT"
        />
      </div>
      <div
        className="col-6"
      >
        <DashboardGraphModule
          benchmarkTitle="benchmark"
          data={null}
          description="How often do you do the following?"
          isFetching={false}
          propertyTitle="property"
          title="While Watching Sports"
          type="PERCENT"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Media Profile Tab should show loading 1`] = `
<div
  className="grid"
>
  <div
    className="breadCrumb"
  >
    <FilterBreadCrumb
      filter={
        Object {
          "benchmark": Object {
            "name": "benchmark",
          },
          "property": Object {
            "name": "property",
          },
        }
      }
    />
  </div>
  <div
    className="container"
  >
    <div
      className="row row-grid"
    >
      <div
        className="col-6"
      >
        <DashboardGraphModule
          benchmarkTitle="benchmark"
          data={null}
          description="Which of the following do you use to interact with or follow the Premier League or your favorite Premier League team?"
          isFetching={true}
          propertyTitle="property"
          sideTitle="Activites"
          title="Interacting with Premier League"
          type="PERCENT"
        />
        <DashboardGraphModule
          benchmarkTitle="benchmark"
          data={null}
          isFetching={true}
          propertyTitle="property"
          title="Sport Watching"
          type="PERCENT"
        />
      </div>
      <div
        className="col-6"
      >
        <DashboardGraphModule
          benchmarkTitle="benchmark"
          data={null}
          description="How often do you do the following?"
          isFetching={true}
          propertyTitle="property"
          title="While Watching Sports"
          type="PERCENT"
        />
      </div>
    </div>
  </div>
</div>
`;
